includes/class-amp-story-post-type.php

```diff
@@ -1553,4 +1637,204 @@ public static function add_new_max_image_size( $image_sizes ) {
+		if ( 'auto-draft' === get_post_status( wp_unslash( $_POST['post_ID'] ) ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => esc_html__( 'Save the AMP story before exporting.', 'amp' ),
+				],
+				403
+			);
+		}
+
+		// Generate and export the archive.
+		$export = self::generate_export( wp_unslash( $_POST['post_ID'] ) );
+
+		// Export failed.
+		if ( is_wp_error( $export ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => $export->get_error_message(),
+				],
+				403
+			);
+		}
+
+		// Failed to export for an unknown reason not related to generating the archive.
+		wp_send_json_error(
+			[
+				'errorMessage' => esc_html__( 'Could not generate the AMP story archive.', 'amp' ),
+			],
+			403
+		);
+	}
+
+	/**
+	 * Generates a Zip archive from the AMP Story.
+	 *
+	 * @param int $post_id The post ID of the AMP Story.
+	 * @return WP_Error
+	 */
+	private static function generate_export( $post_id ) {
+		$post = get_post( $post_id );
+
+		if ( ! $post ) {
+			return new WP_Error( 'amp_story_export_invalid_post', esc_html__( 'The AMP Story does not exist.', 'amp' ) );
+		}
+
+		$slug = sanitize_title( $post->post_title, $post->ID );
+		$file = wp_tempnam( $slug );
+
+		$zip = new ZipArchive();
+		$res = $zip->open( $file, ZipArchive::CREATE | ZipArchive::OVERWRITE );
+
+		// Generate ZIP and send headers.
+		if ( true === $res ) {
+
+			// Create the zip directory.
+			$zip->addEmptyDir( $slug );
+
+			// Add README.txt file.
+			$zip->addFromString( $slug . '/README.txt', 'temporary content...' );
+
+			// Passed to `get_preview_post_link()` for nonce access and to sanitize the output.
+			$query_args = [
+				'story_export' => true,
+				'_wpnonce'     => wp_create_nonce( self::AMP_STORIES_AJAX_ACTION ),
+			];
+
+			// Passed to `wp_remote_get()`.
+			$args = [
+				'cookies'     => wp_unslash( $_COOKIE ), // Pass along cookies so private pages and drafts can be accessed.
+				'timeout'     => 15, // Increase from default of 5 to give extra time for the plugin to identify the sources for any given validation errors; also, response caching is disabled when validating.
+				'sslverify'   => false,
+				'redirection' => 0, // Because we're in a loop for redirection.
+				'headers'     => [
+					'Cache-Control' => 'no-cache',
+				],
+			];
+
+			// Get the preview URL.
+			$response = wp_remote_get( get_preview_post_link( $post, $query_args ), $args );
+
+			// Ensure we have the required data.
+			if ( ! ( is_array( $response ) && isset( $response['body'] ) && isset( $response['headers']['x-amp-export-assets'] ) ) ) {
+				return new WP_Error( 'amp_story_export_response', esc_html__( 'Could not get the AMP Story HTML.', 'amp' ) );
+			}
+
+			// Get the HTML from the response body.
+			$html = $response['body'];
+
+			$export_args = self::get_export_args( $slug );
+
+			if ( $export_args['canonical_url'] ) {
+
+				// Clean up the source.
+				$html = preg_replace( '/(?=<!--)([\s\S]*?-->)/', '', $html );
```

Comment:
What is this doing?

Reply:
Removing the HTML comments

Reply:
I don't think this is necessary, is it?

Reply:
If this were needed, it would be better done in the sanitizer anyway since it could easily locate all the comment nodes via XPath. But we aren't removing comments elsewhere for optimization,  so I think they should be left in place.

Reply:
Well, not necessary but probably good to do since it references the original WordPress install and related URLs. For example, it removes this block.

```
<!--
The style[amp-custom] element is populated with:
    35 B (0%): link#wp-block-library-css[rel=stylesheet][href=http://local.wordpress.test/wp-content/plugins/gutenberg/build/block-library/style.css?ver=1562098564][type=text/css][media=all]
   219 B (25%): link#amp-stories-css[rel=stylesheet][href=http://local.wordpress.test/wp-content/plugins/amp/assets/css/amp-stories.css?ver=1.2.1-alpha][type=text/css][media=all]
     0 B: style#amp-stories-inline-css[type=text/css]
   289 B (25%): link#amp-stories-frontend-css[rel=stylesheet][href=http://local.wordpress.test/wp-content/plugins/amp/assets/css/amp-stories-frontend.css?ver=1.2.1-alpha][type=text/css][media=]
    84 B: amp-story-page#127ad995-f0c1-47a2-81aa-02fef257f373.wp-block-amp-amp-story-page amp-wp-74885c3
    86 B: amp-img.amp-wp-6ed12b8[layout=fill][src=https://localhost/assets/Nature.jpg]
   123 B: div.amp-story-block-wrapper amp-wp-41b864c
    75 B: h1#ab02e26f-4a27-4da4-a00a-6fb09c27a11c.wp-block-amp-amp-story-text amp-wp-f7dbba7
Total included size: 911 bytes (6% of 13,850 total after tree shaking)
-->
```

Reply:
I see. That comment should instead be removed in another way (via #1843). In particular:

```php
add_filter( 'amp_content_sanitizers', function( $sanitizers ) {
    $sanitizers['AMP_Style_Sanitizer']['include_manifest_comment'] = 'never';
    return $sanitizers;
} );
```

That could be added to the same logic that adds the exporter sanitizer in the first place.